operation/fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/client/action/batch/BatchCancelRequestAction.java

```diff
@@ -0,0 +1,432 @@
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpDelete;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.BasicResponseHandler;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.util.EntityUtils;
+
+import com.ibm.fhir.exception.FHIROperationException;
+import com.ibm.fhir.model.type.code.IssueType;
+import com.ibm.fhir.operation.bulkdata.client.action.BulkDataClientAction;
+import com.ibm.fhir.operation.bulkdata.model.JobExecutionResponse;
+import com.ibm.fhir.operation.bulkdata.model.type.JobParameter;
+import com.ibm.fhir.operation.bulkdata.util.BulkDataExportUtil;
+
+/**
+ * Per the specification, after a Bulk Data Request starts, the client may DELETE the request using the Content-Location (polling location)
+ *
+ * <code>
+ * DELETE [polling content location]
+ * Response - Success
+ * HTTP Status Code of 202 Accepted
+ * Optionally a FHIR OperationOutcome resource in the body
+ * Response - Error Status
+ * HTTP status code of 4XX or 5XX
+ * The body SHALL be a FHIR OperationOutcome resource in JSON format
+ * </code>
+ *
+ * @see http://hl7.org/fhir/uv/bulkdata/STU1/export/index.html#bulk-data-delete-request
+ */
+public class BatchCancelRequestAction implements BulkDataClientAction {
+    private static final String CLASSNAME = BatchCancelRequestAction.class.getName();
+    private static final Logger LOG = Logger.getLogger(CLASSNAME);
+
+    private static final BulkDataExportUtil export = new BulkDataExportUtil();
+
+    private static final List<String> INFLIGHT_STATUS = Arrays.asList("STARTING", "STARTED");
+
+    private CloseableHttpClient cli = null;
+    private String batchUrl = null;
+
+    private HttpClientContext ctx = null;
+
+    private Response.Status result = Response.Status.FORBIDDEN;
+
+    @Override
+    public void prepare(CloseableHttpClient cli, String batchUrl, String batchUser, String batchPass) {
+        this.cli = cli;
+
+        // Configure the local context.
+        CredentialsProvider credsProvider = new BasicCredentialsProvider();
+        credsProvider.setCredentials(AuthScope.ANY,
+                new UsernamePasswordCredentials(batchUser, batchPass));
+        ctx = HttpClientContext.create();
+        ctx.setCredentialsProvider(credsProvider);
+
+        this.batchUrl = batchUrl;
+    }
+
+    @Override
+    public Response.Status getResult() {
+        return result;
+    }
+
+    @Override
+    public void close() throws FHIROperationException {
+        if (cli != null) {
+            try {
+                cli.close();
+            } catch (IOException ex) {
+                LOG.throwing(CLASSNAME, "close", ex);
```

Comment:
I see this pattern quite a bit in this PR and wanted to confirm:  throwing is logged at FINER level, meaning that we won't normally see the root exception in the logs... is that ok?
if not, then either
A. these should be logged at INFO or WARNING; or 
B. we should include the exception as the cause of the new OperationException we're building

is there a good reason to avoid B?

Reply:
Right, that's the intent - each is logged at the finer level. In the case of the Close, I opted not to additionally log it out as the Close will then work silently. In many cases, I repack the exception or throw a specific exception in the PR

Reply:
yeah, I wasn't so concerned with losing this particular exception (which i doubt we'll ever hit).  more just a concern with the pattern...I'll go through and double-check the other instances I saw